llarp/path/pathbuilder.cpp

```diff
@@ -429,40 +429,72 @@ namespace llarp
           path_cat, "{} building path -> {} : {}", Name(), path->ShortName(), path->HopsString());
 
       oxenc::bt_list_producer frames;
-
+      std::vector<std::string> frame_str(path::MAX_LEN);
```

Comment:
Medium-ish?:

So this constructor jumped out at me, mostly because I've had to reference vector constructors to double check the 1 and 2 parameter versions and what they do. So [here](https://en.cppreference.com/w/cpp/container/vector/vector) we see that
```C++
std::vector<int> d(3); // creates a 3-element vector holding {0, 0, 0}
```
Which got me thinking, is this vector of strings making a vector of empty strings `path::MAX_LEN` long? Or is it making a vector that could hold strings reserved to size `path::MAX_LEN`? I had a feeling it was the first one, whereas using [`vector::reserve`](https://en.cppreference.com/w/cpp/container/vector/reserve) seems like it wouldn't create new objects in the memory you're reserving.

So I put that shit into godbolt:
```C++
// Method 1:
std::vector<std::string> temp(5);

// Method 2:
std::vector<std::string> temp;
temp.reserve(5);
```
The second method produced non-trivially less instructions than the first; also `.size()` returns 5 for method 1 and 0 for method 2, but `.capacity()` returns 5 for both.

I know this is a weirdly deep dive into one line, but I was curious because this is something I've looked into before in regards to vector constructors

Reply:
> also .size() returns 5 for method 1 and 0 for method 2, but .capacity() returns 5 for both.

That was the intention, as I populate the strings in the vector in reverse order.  Can change to declaring the vector and calling resize on separate lines if it's more readable, but reserve wouldn't be sufficient in this case.